### PR TITLE
Add comments for advanced network in full-example config

### PR DESCRIPTION
**Pull request by blazember**
Advanced network part of https://github.com/hazelcast/hazelcast/issues/14721

- Enhances/fixes some parts of the regular network config as well.
- Refers to the discovery plugin documentations for the proper and up-to-date configuration options

### CODE REVIEW

File: hazelcast/src/main/resources/hazelcast-full-example.xml [hunk 1]
```diff
@@ -389,43 +389,51 @@
...
+                Specifies whether the TCP/IP discovery is enabled or not. Default value is false.
+            - "connection-timeout-seconds":
+                The maximum amount of time Hazelcast is going to try to connect to a well known member
+                before giving up. Setting it to a too low value could mean that a member is not able
+                to connect to a cluster. Setting it to a too high value means that member startup could
+                slow down because of longer timeouts (e.g. when a well known member is not up). Increasing
+                this value is recommended if you have many IPs listed and the members cannot properly
+                build up the cluster. Its default value is 5.
+
+            It has the following sub-elements to fine tune the multicast discovery.
```
Comment (vbekiaris) [inline hazelcast/src/main/resources/hazelcast-full-example.xml]: omit `multicast` here (and in the same passage below)

File: hazelcast/src/main/resources/hazelcast-full-example.xml [hunk 2]
```diff
@@ -519,6 +527,32 @@
...
                 no properties defined by this configuration, Hazelcast will exceptionally try to use the no-arg
                 constructor.
+        * <failure-detector>:
+        A failure detector is responsible to determine if a member in the cluster is unreachable or crashed.
+        Please refer to https://docs.hazelcast.org/docs/latest/manual/html-single/index.html#failure-detector-configuration
+        for the failure detectors implemented by Hazelcast.
+
+        This element has the following sub-element:
+        * <icmp>:
+        ICMP can be used in addition to the other detectors. It operates at layer 3 detects network
```
Comment (vbekiaris) [inline hazelcast/src/main/resources/hazelcast-full-example.xml]: `and detects`

**Comment by mmedenjak** [comment]
run-lab-run

File: hazelcast/src/main/resources/hazelcast-full-example.xml [hunk 3]
```diff
@@ -2540,6 +2574,316 @@
...
+                    Maximum number of times the IP Datagram (ping) can be forwarded, in most cases
+                    all Hazelcast cluster members would be within one network switch/router therefore
+                    default of 0 is usually sufficient.
+
+        * <member-server-socket-endpoint-config>:
+            Configures the server socket used for the member-member communication.
+        * <client-server-socket-endpoint-config>:
+            Configures the server socket used for the member-client communication.
+        * <wan-server-socket-endpoint-config>:
+            Configures the server socket used for the WAN replication.
```
Comment (vbekiaris) [inline hazelcast/src/main/resources/hazelcast-full-example.xml]: Maybe worth to indicate here that more than one WAN server socket can be configured?

File: hazelcast/src/main/resources/hazelcast-full-example.xml (hunk 1 above)
Comment (blazember) [reply hazelcast/src/main/resources/hazelcast-full-example.xml]: Done.

File: hazelcast/src/main/resources/hazelcast-full-example.xml (hunk 2 above)
Comment (blazember) [reply hazelcast/src/main/resources/hazelcast-full-example.xml]: Added and.

File: hazelcast/src/main/resources/hazelcast-full-example.xml (hunk 3 above)
Comment (blazember) [reply hazelcast/src/main/resources/hazelcast-full-example.xml]: Done.

Comment (blazember) [reply hazelcast/src/main/resources/hazelcast-full-example.xml]: This comment revealed https://github.com/hazelcast/hazelcast/issues/14800. On it.

**Comment by blazember** [comment]
Thanks guys for reading this :smile:

**Pull request merged**